src/Microsoft.VisualStudio.Services.Agent/SecretMask.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    public interface ISecretMask
```

Comment:
These are bad names. These classes (ISecretMask, RegexMask, ValueMask, VariableValueMask) do not represent masks. "*******" is the mask. These are the secrets, or secret configuration. Perhaps something like: ISecret, RegexSecret, ValueSecret, VariableSecret.

Really it seems like these classes should be an internal implementation detail of the secret masker class, and the masker class just exposed methods like "AddRegex", "AddValue", "AddVariableName". Then the naming of these classes becomes less significant.